### DISCUSSION TITLE
Support for calculated columns in tables

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2271] [Feature Ideas]
Mathesar should support adding generated columns in tables. Users would set up a formula for how the data is calculated.

See some preliminary research and design work is published in the following wiki pages:

- [Formulas research](https://wiki.mathesar.org/engineering/research/formulas/)

Related features:

- [Allow calculated and dynamic default values](https://github.com/mathesar-foundation/mathesar/discussions/2272)
- [Support for calculated columns to the Data Explorer](https://github.com/mathesar-foundation/mathesar/discussions/2274)

---

*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by jeromeperez** [comment]
Hi
It should be very interesting for example to transform duration (format time 07:45) in a calculated field Nb_hour (7,75) with formula (duration - int(duration)) * 24
